src/api/app/models/bs_request/find_for/query.rb

```diff
@@ -0,0 +1,26 @@
+class BsRequest
+  module FindFor
+    class Query < Base
+      def all
+        # Setup the collection based on params
+        @relation = @relation.with_types(types) if types.present?
+        @relation = @relation.in_states(states) if states.present?
+        if source_project_name.present?
+          @relation = @relation.from_source_project(source_project_name)
+        end
+        if project_name.present?
+          @relation = BsRequest::FindFor::Project.new(@parameters, @relation).all
+        end
+        if user_login.present?
+          @relation = BsRequest::FindFor::User.new(@parameters, @relation).all
+        end
+        if group_title.present?
+          @relation = BsRequest::FindFor::Group.new(@parameters, @relation).all
```

Comment:
Since `BsRequest::FindFor::Project` and `User` and `Group` classes all take a relation and return a relation, couldn't these be made into scopes on the BsRequest model? Then this method could be like:
```ruby
def all
  # Setup the collection based on params
  @relation = @relation.with_types(types) if types.present?
  @relation = @relation.in_states(states) if states.present?
  @relation = @relation.from_source_project(source_project_name) if source_project_name.present?
  @relation = @relation.for_project(@parameters) if project_name.present?
  @relation = @relation.for_user(@parameters) if user_login.present?
  @relation = @relation.for_group(@parameters) if group_title.present?
  @relation = @relation.in_ids(@parameters[:ids]) if @parameters[:ids]
  @relation = @relation.do_search(@parameters[:search]) if @parameters[:search].present?
  @relation
end
```

Reply:
I would also wrap this query in a scope so it would be like:
```ruby
scope :for_user, ->(params) { FindFor::User.new(params).all }
scope :for_group, ->(params) { FindFor::Group.new(params).all }
scope :for_project, ->(params) { FindFor::Project.new(params).all }
scope :find_for, ->(params) { FindFor::Query.new(params).all }
```

And then get rid of all other references to the `FindFor` classes outside of these scopes. That way the rest of the application doesn't need to know about the `FindFor` classes, it just uses the scopes on `BsRequest` model.

Reply:
Makes sense, thanks for the suggestion!